src/server/search/doc_index.cc

```diff
@@ -178,17 +178,16 @@ bool ShardDocIndex::Matches(string_view key, unsigned obj_code) const {
 SearchResult ShardDocIndex::Search(const OpArgs& op_args, const SearchParams& params,
                                    search::SearchAlgorithm* search_algo) const {
   auto& db_slice = op_args.shard->db_slice();
-  auto search_results = search_algo->Search(&indices_);
+  auto search_results = search_algo->Search(&indices_, params.limit_offset + params.limit_total);
```

Comment:
I don't remember the meaning of `limit_offset` and `limit_total` but it _sounds_ like it should be `total - offset` instead of addition. I'll just put it here, you'll know what's the right thing :)

Reply:
No, if we need items 10-15, we need to fetch items 0..15 (or else how would we know what the first 10 are). Thats a big problem of offset based access and why cursors where invented